src/PointwiseFunctions/AnalyticSolutions/GrMhd/SmoothFlow.cpp

```diff
@@ -88,36 +92,17 @@ SmoothFlow::variables(const tnsr::I<DataType, 3>& x, const double t,
       tuples::tagged_tuple_from_typelist<SmoothFlow::dt_variables_t<DataType>>>(
       x, 0.0);
 
-  // Compute the phase: k * (x - vt)
-  const DataType k_dot_x_minus_vt = [&x, &t, this ]() noexcept {
-    auto local_result = make_with_value<DataType>(x, 0.0);
-    for (size_t i = 0; i < 3; ++i) {
-      local_result += gsl::at(wavevector_, i) *
-                      (x.get(i) - make_with_value<DataType>(
-                                      x, t * gsl::at(mean_velocity_, i)));
-    }
-    return local_result;
-  }
-  ();
-
+  const DataType phase = k_dot_x_minus_vt(x, t);
   // Compute the angular frequency: k * v
-  const DataType k_dot_v = [&x, this ]() noexcept {
-    auto local_result = make_with_value<DataType>(x, 0.0);
-    for (size_t i = 0; i < 3; ++i) {
-      local_result += gsl::at(wavevector_, i) *
-                      make_with_value<DataType>(x, gsl::at(mean_velocity_, i));
-    }
-    return local_result;
-  }
-  ();
-
+  const double k_dot_v = std::inner_product(
```

Comment:
how about saving this as a const member variable?  then it could be used on line 48 to initialize `result` making the function simpler.

Reply:
Make it a member variable but don't make it const. We don't want const member variables, as you say.

Reply:
You could also write custom copy and move semantics, but that seems like more work than it's worth for this class :)

Reply:
yeah, just make it non-const